tests/cross/src/test/scala/tests/pc/InlineValueSuite.scala

```diff
@@ -0,0 +1,168 @@
+package tests.pc
+
+import tests.BaseCodeActionSuite
+import munit.TestOptions
+import munit.Location
+
+import scala.meta.internal.metals.TextEdits
+import org.eclipse.{lsp4j => l}
+import scala.meta.internal.jdk.CollectionConverters._
+import scala.meta.internal.metals.CompilerRangeParams
+import java.net.URI
+import scala.meta.internal.mtags.CommonMtagsEnrichments
+import scala.meta.internal.pc.InlineValueProvider.{Errors => InlineErrors}
+
+class InlineValueSuite extends BaseCodeActionSuite with CommonMtagsEnrichments {
```

Comment:
I'm not really sure how to cover such cases correctly. `PcCollector` finds them all and makes no distinction. I guess I could make `PcCollector` remember the structure (tree) of the usages to find the closest definition and its children. Or even better, change how the PcCollector finds matches by making it first find the "definition". This should be probably a solution in the `PcCollector` itself, since currently it also doesn't work properly for highlighting. Any thoughts? Does this work for rename?

Reply:
I think this should work for rename. We find the actual symbol and then compare it, but maybe I am missing something here? But it should work based on PCCollector since you are already finding the clsoes definition within the code action.

Reply:
Well, I am finding the definition but not sending it with the command. Also, it doesn't really matter, 'cause even if I find the correct definition I also have to filter the references to have only the correct ones.
```
object Main {
  val b = {
    val <<a>> = 1
   val b = a + 1
  }
  val a = 3
  val g = a
}
       
```
In such case `PcCollector` finds all the `a` s (which seem to have the same symbol `value a`). You can also see it if you use highlight. It does seem to work for rename but I'm not fully sure why, there is quite a lot of logic in `RenameProvider`, so maybe that's it. Not sure.
I found for myself a workaround, which may does have the best complexity, but I assume the definitions and references found will never be a long list and complexity of #defs x #refs is fine.

Reply:
I've just realised that the behaviour of PcCollector here is quite weird, I will take a look on that

Reply:
It works when we highlight the outer `a`, for inner `a` it highlights all 3 occurrences, renames works always since we don't use PC there. If we wrap it in `def` it always works correctly

Reply:
Yeah... That checks out.

Reply:
Should be fixed by https://github.com/scalameta/metals/pull/4818